src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -38,135 +188,135 @@ export const FileUpload:React.FC<IFileUploadProps>=({
-    const [padding]=useState(10)
-    const [maxHeight,setMaxHeight]=useState<number|undefined>(undefined)
-    const [positionLoading,setPositionLoading]=useState(!isUploading)
-    const [positionTopLoading,setPositionTopLoading]=useState(0)
-    const [opacityLoading,setOpacityLoading]=useState(0)
-    const [positionIsSuccess,setPositionIsSuccess]=useState(!isSuccess)
-    const [opacityIsSuccess,setOpacityIsSuccess]=useState(0)
-    const [loadingContainerHeight,setLoadingContainerHeight]=useState(0)
-    const [widthComponent,setWidthComponent]=useState(0)
-    const [margin]=useState(10)
-    const [widthIsSuccess,setWidthIsSuccess]=useState(0)
-    const [positionIsFailure,setPositionIsFailure]=useState(!isFailure)
-    const [opacityIsFailure,setOpacityIsFailure]=useState(0)
+    const initState:IState={
+        height:undefined,
+        totalHeight:0,
+        totalHeightPlus:0,
+        maxHeight:undefined,
+        padding:10,
+        margin:10,
+        loading:{position:!isUploading,opacity:0},
+        isSuccess:{position:!isSuccess,opacity:0},
+        isFailure:{position:!isFailure,opacity:0},
+        positionTopLoading:0,
+        loadingContainerHeight:0,
+        componentWidth:0,
+        isSuccessWidth:0,
+        isDragEnter:false,
+    }
+    const [state,dispatch]=useReducer(reducer,initState)
 
     // this sets height of the component, is used to transition between heights. 
     useEffect(()=>{
         if(isUploading||isSuccess||isFailure){
-            if(totalHeightPlus){
-                setHeight(totalHeightPlus)
+            if(state.totalHeightPlus){
+                dispatch({type:SET_HEIGHT,value:state.totalHeightPlus})
             }else{
-                setHeight(undefined)
-                setMaxHeight(600)
+                dispatch({type:SET_HEIGHT,value:undefined})
+                dispatch({type:SET_MAXHEIGHT,value:600})
             }
-        }else if(totalHeight){
-            setHeight(totalHeight)   
+        }else if(state.totalHeight){
+            dispatch({type:SET_HEIGHT,value:state.totalHeight})
         }
-    },[isUploading,isSuccess,isFailure,totalHeight,totalHeightPlus])
+    },[isUploading,isSuccess,isFailure,state.totalHeight,state.totalHeightPlus])
 
     // this is to calculate (set) some values
     useEffect(()=>{
         if(containerRef.current?.scrollHeight){
             const contentHeight=containerRef.current.scrollHeight
-            setTotalHeight(contentHeight-padding*2)
-            setMaxHeight(contentHeight-padding*2)
+            dispatch({type:SET_TOTALHEIGHT,value:contentHeight-state.padding*2})
+            dispatch({type:SET_MAXHEIGHT,value:contentHeight-state.padding*2})
         }
         if(rootRef.current?.clientWidth){
-            setWidthComponent(rootRef.current?.clientWidth-margin*2-padding*2)
+            dispatch({type:SET_COMPONENTWIDTH,value:rootRef.current.clientWidth-state.margin*2-state.padding*2})
```

Comment:
I am not the biggest fan of computing values within setting a state or dispatching, declaring your computed value with a named variable then passing that into the dispatch/state gives more information about what is happening.